SparkPost/matchbox
# FE-828 - Update CodeBlock Component

[FE-828](https://jira.int.messagesystems.com/browse/FE-828)

### What Changed
- Updates `<CodeBlock/>` component to use `styled-components`
- Adds new `dark` prop

### How To Test or Verify
- Verify `<CodeBlock/>` renders to match mocks
- Verify new prop works as expected
- Verify unit tests adequately test component behavior
- Verify that passed in child components render code examples as children themselves (see example using a third party syntax highlighting library)

### PR Checklist
- [X] Figure out how to use a third party syntax highlighting library
